vhdl_lang/src/syntax/disconnection.rs

```diff
@@ -0,0 +1,231 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2018, Olof Kraigher olof.kraigher@gmail.com
+
+use super::common::ParseResult;
+use super::expression::parse_expression;
+//use super::separated_list::parse_ident_list;
+use super::tokens::{Kind::*, TokenSpan};
+use crate::ast::token_range::WithTokenSpan;
+use crate::ast::*;
+use crate::syntax::declarative_part::is_recover_token;
+use crate::syntax::subtype_indication::parse_subtype_indication;
+use crate::syntax::Kind;
+use vhdl_lang::syntax::parser::ParsingContext;
+
+/// LRM 7.4 Disconnection Specification
+pub fn parse_disconnection_specification(
+    ctx: &mut ParsingContext<'_>,
+) -> ParseResult<WithTokenSpan<DisconnectionSpecification>> {
+    let start_token = ctx.stream.expect_kind(Disconnect)?;
+    let mut ident = None;
+
+    if let Some(token) = ctx.stream.peek() {
+        match token.kind {
+            Identifier => {
+                // @TODO allow more than 1 ident
```

Comment:
To allow for more than one ident, you could simply use a loop, i.e.,
```rust
idents = Vec::new();
loop {
    idents.push(WithDecl::new(ctx.stream.expect_ident()?));
    if (!ctx.stream.next_kind_is(Comma)) {
        break;
    }
}
```

Keeping track of the choice (i.e., whether it's an identifier, `all` or `others` is going to be a bit annoying, but I'm confident you'll find a solution :)

Reply:
After a long back and forth I think I finally got it.

Reply:
This looks really clean now. Good job!